coralproject/talk
# Error handling improvements

## What does this PR do?
- https://www.pivotaltracker.com/story/show/154475907
-`withQuery` and `withMutation` now produces an HOC that supports a parameter `notifyOnError: boolean` which defaults to true. If `notifyOnError` is true, a notification is shown to the user when the mutation or query has failed.

```js
// Example usage.
const withBanUser = withMutation(...);

const enhance = compose(
  withBanUser({notifyOnError: false})
);

export default enhance(MyComponent);
``` 

- Major error handling refactor to notify the user whenever an error has happened
- Stories on Admin Page has not been modified as it still relies on REST calls
- Improve error handling of report username / comment feature (https://www.pivotaltracker.com/story/show/154570331)

## How do I test this PR?

- Report username / comment feature 
  - Report a comment, but turn off network in devtools, to make it fail. The comment should not be marked as reported nor should a `thank you` dialog be shown

- Rest
  - Try to fail any action (except the Stories Page) and a notification should be visible to alert the user that something has gone wrong. You can turn off and on the network in devtools to cause an error.
